packages/amplify_api/lib/method_channel_api.dart

```diff
@@ -159,8 +160,8 @@ class AmplifyAPIMethodChannel extends AmplifyAPI {
             AmplifyExceptionMessages.nullReturnedFromMethodChannel);
       final errors = _deserializeGraphQLResponseErrors(result);
 
-      GraphQLResponse<T> response =
-          GraphQLResponse<T>(data: result['data'] ?? '', errors: errors);
+      GraphQLResponse<T> response = GraphQLResponseDecoder.instance
```

Comment:
I don't see any tests for the decoder's integration in the method channel. If you look at amplify_api_query_test.dart you will see testing for how the api plugin itself responds in certain situations (given that the JSON response [or error] from method channels is mocked). This would be a good place to add at least 1 test for this. You could pretty much copy/paste the first test, but produce the request from your helper, then test that the response is decoded, similar to existing test, just that this would ensure the methods work as expected in the plugin itself, rather than when calling internal helpers directly.